lib/entice/logic/vitals.ex

```diff
@@ -4,61 +4,139 @@ defmodule Entice.Logic.Vitals do
+                   |> update_attribute(Energy, fn _ -> %Energy{mana: resurrected_mana, max_mana: max_mana} end)}
     end
 
+    def init(%Entity{attributes: %{Level => %Level{level: _}}} = entity, _args) do
+      {:ok, entity |> put_attribute(get_max_health(entity))
+                   |> put_attribute(get_max_energy(entity))
+                   |> put_attribute(%Morale{morale: 0})}
+    end
+
+    def terminate(:remove_handler, entity) do
+      {:ok, entity}
+    end
 
     def terminate(_reason, entity) do
       {:ok, entity |> remove_attribute(Health)
-                   |> remove_attribute(Energy)}
+                   |> remove_attribute(Energy)
+                   |> remove_attribute(Morale)}
+    end
+
+    def handle_event({:vitals_entity_damage, amount}, %Entity{attributes: %{Health => %Health{health: health, max_health: _}}} = entity) do
+      new_health = health - amount
+      cond do
+        new_health <= 0 -> {:become, Vitals.DeadBehaviour, {entity, :vitals_entity_died}, entity |> update_attribute(Health, fn health -> %Health{health: 0, max_health: health.max_health} end)}
+        new_health > 0 -> {:ok, entity |> update_attribute(Health, fn health -> %Health{health: new_health, max_health: health.max_health} end)}
+      end
     end
 
+    def handle_event({:vitals_entity_heal, amount}, %Entity{} = entity) do
+      {:ok, %Health{health: health, max_health: max_health}} = fetch_attribute(entity, Health)
+      new_health = health + amount
+      if new_health > max_health do
+        new_health = max_health
+      end
 
-    # internal
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: new_health, max_health: max_health} end)}
+    end
 
+    # internal
 
     defp get_max_health(entity) do
       {:ok, level} = fetch_attribute(entity, Level)
+      {:ok, morale} = fetch_attribute(entity, Morale)
       health = calc_life_points_for_level(level.level)
-      %Health{health: health, max_health: health}
+      max_health_with_morale = health / 100 * (100 + morale.morale)
+      %Health{health: max_health_with_morale, max_health: max_health_with_morale}
     end
 
-
     #TODO: Take care of Armor, Runes, Weapons...
     defp calc_life_points_for_level(level),
     do: 100 + ((level - 1) * 20) # Dont add 20 lifePoints for level1
 
-
     #TODO: Take care of Armor, Runes, Weapons...
-    defp get_max_energy(_entity),
-    do: %Energy{mana: 70, max_mana: 70}
+    defp get_max_energy(entity) do
+      {:ok, morale} = fetch_attribute(entity, Morale)
+      inital_mana = 70
+      mana_with_morale = inital_mana / 100 * (100 + morale.morale)
+      %Energy{mana: mana_with_morale, max_mana: inital_mana}
+    end
+  end
+
+  defmodule DeadBehaviour do
+    use Entice.Entity.Behaviour
+    alias Entice.Logic.Vitals.Morale
+
+    def init(%Entity{attributes: %{Morale => %Morale{morale: morale}}} = entity, :vitals_entity_died) do
```

Comment:
also we need to be aware that this means semantically that there is no way to init an entity as being dead (except if you know that your entity needs to have a morale and you need to pass in `:entity_died` as an arg, which usually only this module knows)
ofc, the question is if we will ever need to init an entity as being dead, and in that case we can probably refactor this, so, nevermind ;D